tempodb/encoding/vparquet/wal_block.go

```diff
@@ -642,13 +645,17 @@ func (b *walBlock) Fetch(ctx context.Context, req traceql.FetchSpansRequest, opt
 
 		wrappedIterator := &pageFileClosingIterator{iter: iter, pageFile: file}
 		iters = append(iters, wrappedIterator)
+		totalBytesRead += file.r.TotalBytesRead.Load()
 	}
 
 	// combine iters?
 	return traceql.FetchSpansResponse{
 		Results: &mergeSpansetIterator{
 			iters: iters,
 		},
+		Bytes: func() uint64 {
+			return totalBytesRead
```

Comment:
I'm not sure this will work.  Fetch just opens the iters, so the value `totalBytesRead` just counts the parquet headers more or less.  Since `mergeSpansetIterator` is the top-level iter returned to the engine, it would need to sum up these values when each iter is finished (or engine exits early).

Reply:
updated, now I am collecting readers from pages and reading TotalBytesRead when Bytes callback is called.

this means, that we will read metadata + data size post iteration, and it will produce accurate result.